### DISCUSSION TITLE
Thanks from the FFL team Exploding Toaster Supernova

### DISCUSSION THREAD

**Discussion by thehomelessguy** [pybricks/support#999] [Show and tell]
We are a first year FLL team who was struggling at the end of December. At the last scrimage the kids scored only 175 and were a bit discouraged. We had a mess of scratch/blockly style code that as my son described it "is psychedelic he**". We made the decision to transition to legacy lego python however I could not get Legos firmware downgrader to work. I actually stumbled upon Pybricks as you folks had a tutorial to get the firmware update working (it was the driver issue) that came up in our google searches. 

Anyways my son and I checked out the rest of the website and were intrigued by things like the remote code (we immediately had visions of Robot Fight Club) and went with Pybricks instead. Truely a game changer. It has only been a bit over two months since they switched to Pybricks, but they went from scoring 175 in scrimmage to scoring 375 (They had 400 points coded but things were not 100% reliable). They even rebuilt all of our past robots (using cheap technic hubs rather than spike ones) and did a little synchronized robot parade using the experimental broadcast code as part of their robot design presentation. Thanks to laurensvalk for providing a newly compiled firmware for such. 

Sadly we did not really understand the other parts of the FLL challenge (our presentation was apparently not up to snuff despite the parade) and will not be moving forward despite our very dominant robot game performance. We were looking forward to the possibility of meeting some other Pybricks users at worlds/post-season/etc. However the kids were very happy with the trophy and are super excited about next year.

Huge thanks to all the developers. You have created a fantastic community and product. Looking forward to Pybricks continued development and will start providing support for such. We may even release some examples of how we added the ability to handle multiple programs/update the hub display/ and the other code needed to get up and running in Pybricks as I would love for other teams to experience the improvements Pybricks offers. My son and I are also exploring how hard it would be to create a blockly/scratch style layer so that things like the DriveBase module and the much more accurate movement and ease of programming that offers could be used by teams not yet able to use text-based code. May take a while to figure out as we are very much learners in this area/not developers(yet) (I used to write a lot of code in LISP for computational modeling of cognition and do some stuff in R for analyzing research, but that is pretty much the extent of my experience, and my son only has myself as a teacher). 

Thanks again! Here is our highest scoring run (Exploding Toaster Supernova, Table D1, right table on the bottom left screen wearing the multicolored fedoras). Would not have happened without your generosity.
https://youtu.be/I4g1R-PBdCY?t=16500

**Comment by dlech** [comment]
Thanks for sharing you experience. I would be interested in seeing your team's scratch code compared to the Python code.

**Comment by TheWendyPower** [comment]
Our team would love to help you out with presentations next year!
